qutebrowser/browser/commands.py

```diff
@@ -955,7 +955,7 @@ def buffer(self, index):
 
     @cmdutils.register(instance='command-dispatcher', scope='window')
     @cmdutils.argument('index', choices=['last'])
-    @cmdutils.argument('count', count=True)
+    @cmdutils.argument('count', count=True, zero_count=True)
     def tab_focus(self, index: typing.Union[str, int]=None, count=None):
```

Comment:
Just to reiterate on this again, because I think this is confusing you: `:tab-focus` can accept a tab number either as argument `index` (like `:tab-focus 1`) or as a count via a keybinding (e.g. when pressing `0T`, then `index` would be `None` but `count` would be `0`).

Then the `index = count if count is not None else index` line below overrides the index with the count if it's given.